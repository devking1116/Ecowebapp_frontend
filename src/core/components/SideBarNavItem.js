/* @flow */
import React from 'react';
import reactCSS from 'reactcss';
import { Popup  } from 'semantic-ui-react';

type Props = {
  label: string,
  icon: any,
  isExpanded?: bool,
};

/**
 * A list item that can be displayed in expanded or collapsed formate, for side bars used for navigation.
 */
export default function SideBarNavItem({ label, icon, isExpanded, ...otherProps }: Props): React.Element<*> {
  if(isExpanded) {
    const styles = reactCSS({
      'default': {
        label: {
          paddingLeft: 25,
        },
      },
    });

    return (
      <span {...otherProps}>
        {icon}
        <span style={styles.label}>{label}</span>
      </span>
    );
  }
  else {
    return (
      <Popup
        trigger={<span {...otherProps}>{icon}</span>}
        content={label}
        positioning="right center"
      />
    );
  }
}
