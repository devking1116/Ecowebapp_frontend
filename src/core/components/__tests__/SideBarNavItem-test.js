import React from 'react';
import renderer from 'react-test-renderer';

import SideBarNavItem from '../SideBarNavItem';

test('when expanded, it displays its label next to its text.', () => {
  const component = renderer.create(
    <SideBarNavItem isExpanded label="Test Label" icon={<i className="test-icon" />} />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('when not expanded, it just displays an icon and puts the label in a popup.', () => {
  const component = renderer.create(
    <SideBarNavItem label="Test Label" icon={<i className="test-icon" />} />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
