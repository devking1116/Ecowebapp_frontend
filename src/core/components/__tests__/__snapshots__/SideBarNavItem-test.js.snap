exports[`test when expanded, it displays its label next to its text. 1`] = `
<span>
  <i
    className="test-icon" />
  <span
    style={
      Object {
        "paddingLeft": 25,
      }
    }>
    Test Label
  </span>
</span>
`;

exports[`test when not expanded, it just displays an icon and puts the label in a popup. 1`] = `
<span
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  <i
    className="test-icon" />
</span>
`;
